utils/http/retry_test.go

```diff
@@ -0,0 +1,94 @@
+package http
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/atomic"
+
+	"github.com/ARM-software/golang-utils/utils/commonerrors"
+	"github.com/ARM-software/golang-utils/utils/commonerrors/errortest"
+	"github.com/ARM-software/golang-utils/utils/logs/logstest"
+)
+
+func TestRetryOnError(t *testing.T) {
+	tests := []struct {
+		policy *RetryPolicyConfiguration
+	}{
+		{
+			policy: DefaultBasicRetryPolicyConfiguration(),
+		},
+		{
+			policy: DefaultNoRetryPolicyConfiguration(),
+		},
+		{
+			policy: DefaultLinearBackoffRetryPolicyConfiguration(),
+		},
+		{
+			policy: DefaultExponentialBackoffRetryPolicyConfiguration(),
+		},
```

Comment:
also I know it is annoying, but I think we should have a cancelled context test. The `.Context(ctx)` should handle this, but in the event we switch implementations one day it would be good to have a test to make sure we account for cancelled contexts. Especially since cancelling http contexts seems like it is a common thing that might occur